encoding/delta/binary_packed_decode_amd64.s

```diff
@@ -134,6 +135,304 @@ test:
     JNE loop
     RET
 
+// -----------------------------------------------------------------------------
+// The decodeMiniBlock* functions below are adaptations of the algorithms
+// described in "Decoding billions of integers per second through vectorization"
+// from D. Lemire & L. Boytsov, the following changes were made:
+//
+// - The paper described two methods for decoding integers called "horizontal"
+//   and "vertical". The "horizontal" version is the one that applies the best
+//   to the bit packing done in the Parquet delta encoding; however, it also
+//   differs in some ways, many compression techniques discussed in the paper
+//   are not implemented in the Parquet format.
+//
+// - The paper focuses on implementations based on SSE instructions, which
+//   describes how to use PMULLD to emulate the lack of variable bit shift
+//   for packed integers. Our version of the bit unpacking algorithms here
+//   uses AVX2 and can perform variable bit shifts using VPSRLVD, which yields
+//   better throughput since the instruction latency is a single CPU cycles,
+//   vs 10 for VPMULLD.
+//
+// - The reference implementation at https://github.com/lemire/FastPFor/ uses
+//   specializations for each bit size, resulting in 32 unique functions.
+//   Our version here are more generic, we provide 3 declinaisons of the
```

Comment:
```suggestion
//   Our version here is more generic, we provide 3 declarations (?) of the
```

Reply:
French to English dictionary also suggests "declensions", either fine with me.

Reply:
I think what I meant was "variations" 👍